frontend/app/forms/application-for-pardon/application-for-pardon.web.component.js

```diff
@@ -0,0 +1,202 @@
+
+export default function ApplicationForPardon_Web({ data }) {
+  return (
+    <Scoped css={css}>
+      <FormThatPrints>
+        <Section name="1. Personal information">
+          <TextInput dataKey="person.firstName" label={__("first name")} />
+          <TextInput dataKey="person.middleName" label={__("middle name")} />
+          <TextInput dataKey="person.lastName" label={__("last name")} />
+          <TextInput dataKey="person.birthday" label={__("birthday")} />
+          <TextInput
+            dataKey="person.previouslyUsedNames"
+            label={__("previously used names")}
+          />
+          <TextInput dataKey="person.addressStreet" label={__("street")} />
+          <TextInput dataKey="person.addressCity" label={__("city")} />
+          <TextInput dataKey="person.addressState" label={__("state")} />
+          <TextInput dataKey="person.addressZip" label={__("zip")} />
+          <TextInput
+            dataKey="person.socialSecurity"
+            label={__("social security number")}
+          />
+          <TextInput
+            dataKey="person.driversLicenseNumber"
+            label={__("drivers license #")}
+          />
+          <TextInput
+            dataKey="person.driversLicenseState"
+            label={__("drivers license state")}
+          />
+          <TextInput
+            dataKey="person.driversLicenseState"
+            label={__("drivers license state")}
+          />
+          <TextInput
+            dataKey="person.offenderNumber"
+            label={__("offender #")} // Offender #
+          />
+          <TextInput
+            dataKey="person.fbiNumber"
+            label={__("fbi #")} // FBI #
+          />
+          <TextInput dataKey="person.homePhone" label={__("home phone")} />
+          <TextInput dataKey="person.dayPhone" label={__("day phone")} />
+          <TextInput dataKey="person.email" label={__("email address")} />
+        </Section>
+
+        <Section name="2. Pardon Request and Consideration">
+          <Radio
+            dataKey="person.previouslyApplied"
+            label={__("previously applied")}
+            options={previouslyAppliedOptions}
+          />
+
+          <p className="web-form-input">
+            If yes, please provide the date of the application, the outcome of
+            the application, list any conviction(s) pardoned, and attach a copy
+            of the application and the Pardon Certificate.
+          </p>
+          {data.person.previouslyApplied && (
+            <TextArea
+              dataKey="person.explaination" // needs more specific name
+              label={__("previously please explain")}
+            />
+          )}
+        </Section>
+        <Section name="3. Family History">
+          <p className="web-form-input">
+            If married, please provide the following information for your spouse
+          </p>
+          <TextInput
+            dataKey="spouse.firstName"
```

Comment:
Is `spouse.firstName` a new data key?

The `.` in data keys represents a sub object of a collection in firebase. We have two modes for the data entry - first is just in-memory store and the second is with a login via firebase.

Right now we only have two collections - `case` and `person`. We allow users to switch which case and which person (in the case they're a lawyer they have multiple clients) they're working on. By introducing a new top-level collection `spouse`, you're (probably inadvertently) indicating that spouse is also something that could be swapped out independent of the person. Which isn't something that we have a UI for, nor is it a feature that we want.

Instead, this should be `person.spouseFirstName` or something similar, just tacked on as a property on the `person` collection.

Reply:
Yes, it is a new data key. 

Yep, sounds good. I made this commit when I didn't know what structure our data was.